pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -570,6 +575,47 @@ func GetVirtualCenterInstance(ctx context.Context,
 	return vCenterInstance, nil
 }
 
+// GetVirtualCenterInstanceForVCenterHost returns the vcenter object for given vCenter host
+// Takes in a boolean paramater reloadConfig.
+// If reinitialize is true, the vcenter object is instantiated again and the
+// old object becomes eligible for garbage collection.
+// If reinitialize is false and instance was already initialized, the previous
+// instance is returned.
+func GetVirtualCenterInstanceForVCenterHost(ctx context.Context,
+	vcconfig *VirtualCenterConfig, vcHost string, reinitialize bool) (*VirtualCenter, error) {
+	log := logger.GetLogger(ctx)
+	vCenterInstancesLock.Lock()
+	defer vCenterInstancesLock.Unlock()
+
+	_, found := vCenterInstances[vcHost]
+	if !found || reinitialize {
+		log.Infof("Initializing new vCenterInstance for vCenter %q", vcHost)
+		// Initialize the virtual center manager.
+		virtualcentermanager := GetVirtualCenterManager(ctx)
+		// Unregister VCs from virtual center manager.
```

Comment:
`//  Unregister VCs ` -> `//  Unregister the VC`
as we are not unregistering all VCs here

Reply:
done.